File: application/backend-credit-card-enrollment/backend-java/src/main/java/cloud/ambar/common/reaction/ReactionHandler.java
package cloud.ambar.common.reaction;

import cloud.ambar.common.event.Event;
import cloud.ambar.common.eventstore.PostgresTransactionalEventStore;
import com.fasterxml.jackson.core.JsonProcessingException;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public abstract class ReactionHandler {
    final protected PostgresTransactionalEventStore postgresTransactionalEventStore;
    protected abstract void react(final Event event) throws JsonProcessingException;
}

File: application/backend-credit-card-enrollment/backend-c#/Common/Reaction/ReactionHandler.cs
using CreditCardEnrollment.Common.EventStore;

namespace CreditCardEnrollment.Common.Reaction;

public abstract class ReactionHandler {
    protected readonly PostgresTransactionalEventStore _postgresTransactionalEventStore;

    protected ReactionHandler(PostgresTransactionalEventStore postgresTransactionalEventStore) {
        _postgresTransactionalEventStore = postgresTransactionalEventStore;
    }

    public abstract void React(Event.Event @event);
}